src/plugins/service-worker/index.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+const buildPaths = require('mwp-cli/src/commands/buildCommands/config/paths');
+const path = require('path');
+
+/*
+ * The service worker plugin will register a root-level route for serving the
+ * service worker script file.
+ *
+ * Disk location of the script file is determined by builder CLI config import.
+ */
+export default function register(
+	server: Object,
+	options: ?Object,
+	next: () => void
+) {
+	/*
+	 * Route for service worker script at top-level path. Depends on `Inert`
+	 * `path` must match client `serviceWorker.register` call - MWP provides this
+	 * in the `<ServiceWorker>` component
+	 */
+	server.route({
+		method: 'GET',
+		path: '/asset-service-worker.{localeCode}.js',
```

Comment:
> caching behavior is implementation specific.

I'm not sure what this means. The precache behavior is described here https://github.com/GoogleChrome/sw-precache, but essentially the service worker will download _all_ the files generated by the webpack build in the background when it first installs. This means that async chunks and some of the images that are generated by the bundle will load instantly when they're needed on subsequent navigation.

One of my tickets for this sprint is to document current caching behavior in the platform - I'll make sure to include a section about this service worker.